internal/olm/operator/registry/configmap/configmap_test.go

```diff
@@ -0,0 +1,277 @@
+						},
+						{
+							Object: map[string]interface{}{"val2": "va2"},
+						},
+					},
+					CSV: &v1alpha1.ClusterServiceVersion{
+						Spec: v1alpha1.ClusterServiceVersionSpec{
+							Version: version.OperatorVersion{
+								Version: semver.SpecVersion,
+							},
+						},
+					},
+				},
+				{
+					Name: "testbundle_2",
+					Objects: []*unstructured.Unstructured{
+						{
+							Object: map[string]interface{}{"val1": "val1"},
+						},
+						{
+							Object: map[string]interface{}{"val2": "va2"},
+						},
+					},
+					CSV: &v1alpha1.ClusterServiceVersion{
+						Spec: v1alpha1.ClusterServiceVersionSpec{
+							Version: version.OperatorVersion{
+								Version: semver.SpecVersion,
+							},
+						},
+					},
+				},
+			}
+
+			p := apimanifests.PackageManifest{
+				PackageName: "test_package_manifest",
+				Channels: []apimanifests.PackageChannel{
+					{Name: "test_1",
+						CurrentCSVName: "test_csv_1",
+					},
+					{Name: "test_2",
+						CurrentCSVName: "test_csv_2",
+					},
+				},
+				DefaultChannelName: "test_channel_name",
+			}
+
+			binaryDataByConfigMap, err := makeConfigMapsForPackageManifests(&p, b)
+
+			val := make(map[string]map[string][]byte)
+			cmName := getRegistryConfigMapName(p.PackageName) + "-package"
+			val[cmName], err = makeObjectBinaryData(p)
+			for _, bundle := range b {
+				version := bundle.CSV.Spec.Version.String()
+				e = fmt.Errorf("bundle ClusterServiceVersion %s has no version", bundle.CSV.GetName())
+
+				cmName := getRegistryConfigMapName(p.PackageName) + "-" + k8sutil.FormatOperatorNameDNS1123(version)
+				val[cmName], e = makeBundleBinaryData(bundle)
+			}
+
+			Expect(e).Should(BeNil())
+			Expect(err).Should(BeNil())
+			Expect(binaryDataByConfigMap).Should(Equal(val))
+
+		})
+
+	})
+
+	Describe("getRegistryConfigMaps", func() {
+		It("performs operations and returns all the configmaps", func() {
+			var fclient RegistryResources
+			fakeclient := fake.NewFakeClient(
+				&corev1.ConfigMapList{
+					Items: []corev1.ConfigMap{
+						{
+							ObjectMeta: metav1.ObjectMeta{
+								Namespace: "testns",
+								Labels:    makeRegistryLabels("test"),
+							},
+						},
+						{
+							ObjectMeta: metav1.ObjectMeta{
+								Namespace: "testns2",
+								Labels:    makeRegistryLabels("test"),
+							},
+						},
+					},
+				},
+			)
+			fclient = RegistryResources{
+				Client: &client.Client{
+					KubeClient: fakeclient,
+				},
+				Pkg: &apimanifests.PackageManifest{
+					PackageName: "test",
+				},
+				Bundles: fclient.Bundles,
+			}
+
+			list := corev1.ConfigMapList{}
+			opts := []client_cr.ListOption{
+				client_cr.MatchingLabels(makeRegistryLabels(fclient.Pkg.PackageName)),
+				client_cr.InNamespace("testns"),
+			}
+			e := fclient.Client.KubeClient.List(context.TODO(), &list, opts...)
+			configmaps, err := fclient.getRegistryConfigMaps(context.TODO(), "testns")
+
+			Expect(e).Should(BeNil())
+			Expect(err).Should(BeNil())
```

Comment:
Same comment as all the others, put these 2 error Expectations closer to where the problem happens. The only reason I move them is because the errors are part of the SETUP not really the thing we're testing.